packages/react-router/src/ReactRouter/StackManager.tsx

```diff
@@ -206,8 +206,18 @@ export class StackManager extends React.PureComponent<StackManagerProps, StackMa
   registerIonPage(page: HTMLElement, routeInfo: RouteInfo) {
```

Comment:
Is there any way to move the checks here up into the calling components, so that `registerIonPage` is not invoked multiple times?

The React 18 change was designed to help developers migrate problematic life cycle method implementations that produce side effects, as they prepare for larger changes in React 19 to how components are attached/re-attached in the DOM (for optimization benefits). I believe the intention is that the dev-mode behavior will become the default production behavior in one of the upcoming majors React 19/20. 

My only concern here with fixing in this API method, is that technically the problematic pattern still exists in the components:
- https://github.com/ionic-team/ionic-framework/blob/main/packages/react/src/routing/PageManager.tsx#L33
- https://github.com/ionic-team/ionic-framework/blob/main/packages/react/src/routing/OutletPageManager.tsx#L29

while the behavior has obviously been patched in this specific method. Future updates to `PageManager`/`OutletPageManager`, would not have the immediate awareness that additional method calls in `componentDidMount` will lead to problematic behavior. 

If there isn't a tangible way to move this up into the component life cycle calls, then I would be in favor of this change as-is.

Reply:
How would you recommend I change the code? I tried a couple alternatives already:
1. Moving the logic to `constructor` will not work as the `this.ionPageElementRef.current` may not be available at that point. 
2. I could add a "init" flag on each component so that subsequent `componentDidMount` calls do not re-fire `registerIonPage`, but that seems a bit hacky.

Reply:
Great question... not entirely sure myself. I've seen the `init` flag being used in a few places before, but agree it is a bit hacky. 

I agree with your point that `registerIonPage` should be idempotent & works great for `PageManager`. The only downside with the current implementation is within `OutletPageManager`. Since `componentOnReady` will be invoked multiple times; it will end up calling `requestAnimationFrame` and tying up the repaint. 

I could see the justification in that component alone, to have a flag set for when the component is "ready" and checking if that flag has already been set before invoking that call again. This would prevent unnecessary `raf` calls.

Thoughts?

Reply:
That should be fine. I made the changes in https://github.com/ionic-team/ionic-framework/pull/25798/commits/0458734308c8722fc9e9e48f3089b8eda1ea5bb6.

Reply:
Ok tested with React 17 and React 18 and works fine 👍